### ISSUE TITLE
"error while loading shared libraries" while browsing the vfs

### ISSUE THREAD

**Issue by saviola777** [oniony/TMSU#52]
Hello,

here's a minimal shell session to reproduce and illustrate the problem (on my machine) using the latest stable release:

```
saviola@trinity ~/tmp/test $ tmsu init
tmsu: creating database at '/home/saviola/tmp/test/.tmsu/db'.
saviola@trinity ~/tmp/test $ mkdir tmsu            
saviola@trinity ~/tmp/test $ tmsu -D .tmsu/db mount tmsu
saviola@trinity ~/tmp/test $ touch file
saviola@trinity ~/tmp/test $ tmsu -D .tmsu/db tag file tag1
tmsu: New tag 'tag1'.
saviola@trinity ~/tmp/test $ cd tmsu/tags/tag1
/usr/bin/python3.4: error while loading shared libraries: libpython3.4.so.1.0: file too short
saviola@trinity ~/tmp/test/tmsu/tags/tag1 $ ls -lA
total 0
lrwxr-xr-x 1 saviola saviola 0 Oct 21 14:32 file.1 -> /home/saviola/tmp/test/file
saviola@trinity ~/tmp/test/tmsu/tags/tag1 $ ....
saviola@trinity ~/tmp/test $ mv file file1
saviola@trinity ~/tmp/test $ cd tmsu/tags/tag1 
saviola@trinity ~/tmp/test/tmsu/tags/tag1 $ ls -lA
total 0
lrwxr-xr-x 1 saviola saviola 0 Jan  1  0001 file.1 -> /home/saviola/tmp/test/file
```

So in short: When browsing the vfs, I get errors only when the linked file actually exists, otherwise it works fine. The error appears after every change of directory below the "tags" directory (and when entering directories in the queries directory)  in the vfs. It may be worth noting that with the current stable release, I get an additional error when trying to list the files ("error while loading shared libraries: libacl.1.so: file too short" or something very similar), this seems to be fixed in the current master, at least making TMSU usable for me. Also, I cannot reproduce the issue on my notebook with the current stable release. Since both systems are similar, I will try to look for version differences in the installed software to identify the culprit, is there some easy way to track down all dependencies of TMSU (and their dependencies, and so on) to narrow things down a bit?

As I said, this does not break TMSU for me, but it may be worth looking into, if and when you have some time (may be another upstream issue, I don't know). Thanks, and keep up the good work.

**Comment by 0ion9** [comment]
Whenever I get a 'error while loading shared libraries', I generally find that a program is out of date (a library has been updated but the program has not been recompiled to use the new version).

Anyway, TMSU does not directly use python at all (`grep -ri python *` in the TMSU sourcetree returns nothing), nor does FUSE core, so I'm pretty sure this will turn out to be a problem with a dependency.

**Comment by saviola777** [comment]
I have narrowed it down to the combination of tmux + zsh. When using only zsh, no error. When using tmux + bash, no error. Only when using tmux + zsh do I get the above error… But I have no idea which of them uses python. I guess this will be a little difficult to track down, and is most probably not a TMSU error, so I will close this and report back if and when I found a solution or cause. Thanks!
